### PR TITLE
london 8/ margarita glushkova / JavaScript 1 week 1

### PR DESCRIPTION
**Pull request by RitaGlushkova**
All mandatory exercises are complete and tested.

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Comment by RitaGlushkova** [comment]
@DiegoJunior0 Hi Henry, my mandatory exercises are complete and tested.

**Comment by RitaGlushkova** [comment]
@DiegoJunior0 Hi Henry all exercises are complete including extra. But 3-magic-8 ball only works if I use Run from vs code and comment //test section. If I try to test it with test code - it fails.

File: extra/2-piping.js [hunk 1]
```diff
@@ -16,27 +16,25 @@
...
+function format(num) {
+  return "£" + num;
 }
 
 const startingValue = 2;
 
 // Why can this code be seen as bad practice? Comment your answer.
-let badCode =
-
+let badCode = format(multiply(add(startingValue, 10), 2));
```
Comment (DiegoJunior0) [inline extra/2-piping.js:34]: I believe the idea here is to do the multiply and add on separate lines using variables to store the values. This makes the code easier to read and to debug.

File: mandatory/3-function-output.js [hunk 2]
```diff
@@ -2,13 +2,16 @@
 function getRandomNumber() {
   return Math.random() * 10;
 }
+//A floating-point, pseudo-random number between 0 (inclusive) and 1 (exclusive)
 
 // Add comments to explain what this function does. You're meant to use Google!
 function combine2Words(word1, word2) {
-  return word1.concat(word2);
+  return word1.concat("", word2);
```
Comment (DiegoJunior0) [inline mandatory/3-function-output.js:9]: this addition of a empty string as the first parameter of concat is un-necessary. `word1.concat(word2)` works fine.

**Comment by DiegoJunior0** [comment]
Excellent work Margarita, could only see one minor issue

**Pull request closed without merging**